File: backend_spring/tass-spring/src/main/java/pl/tass/tassspring/model/dto/graph/GraphDTO.java
package pl.tass.tassspring.model.dto.graph;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import pl.tass.tassspring.model.dto.NetworkPropDTO;

import java.util.List;

@AllArgsConstructor
@Data
@NoArgsConstructor
@Builder
public class GraphDTO {
    private List<GraphNodeDTO> nodes;
    private List<GraphLinkDTO> links;
    private NetworkPropDTO networkProp;
}

File: frontend/src/app/shared/model/dto/graph/graph.dto.ts
import { GraphNodeDto } from './graph-node.dto';
import { GraphLinkDto } from './graph-link.dto';
import { NetworkPropDto } from '../network-prop.dto';

export interface GraphDto {
  nodes: GraphNodeDto[];
  links: GraphLinkDto[];
  networkProp: NetworkPropDto;
}
